pootle/apps/virtualfolder/helpers.py

```diff
@@ -46,3 +46,11 @@ def extract_vfolder_from_path(request_path):
         return None, request_path
     else:
         return vfti.vfolder, vfti.directory.pootle_path
+
+
+def vftis_for_children(directory):
+    """
+    Returns the vfoldertreeitems for a directory's children
+    """
+    child_pks = [child.pk for child in directory.children]
+    return VirtualFolderTreeItem.objects.filter(directory__pk__in=child_pks)
```

Comment:
With the code above it seems that you are retrieving the PKs for all children (stores and subdirs) but then retrieving the vftis assuming they are only for directories?